src/v/kafka/server/connection_context.cc

```diff
@@ -36,16 +38,30 @@
 #include <seastar/core/scattered_message.hh>
 #include <seastar/core/shared_ptr.hh>
 #include <seastar/core/sleep.hh>
+#include <seastar/core/temporary_buffer.hh>
 #include <seastar/core/with_timeout.hh>
 #include <seastar/coroutine/as_future.hh>
 
 #include <chrono>
+#include <cstdint>
 #include <memory>
 
 using namespace std::chrono_literals;
 
 namespace kafka {
 
+namespace {
+static constexpr std::string_view
+  multi_proxy_initial_client_id("__redpanda_mpx");
+
+bytes parse_virtual_connection_id(const ss::temporary_buffer<char>& buffer) {
+    // TODO: should we use vcluster_id here ?
+    return bytes{
+      reinterpret_cast<const uint8_t*>(buffer.begin()), buffer.size()};
+}
```

Comment:
question: So once we get the 'magic' `mpx` value, each subsequent request will have the virtual connection ID in the client_id.  Is there any format to that value?  Does it need any additional validation or once we know we're a virtual connection, any value for client ID is accepted?

Reply:
we haven't yet agreed with the team about the format, i made it to accept any string for now, a plan is to add schema to it in future PR